akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -323,6 +323,11 @@ object Source extends SourceApply {
    * Every connected `Sink` of this stream will see an individual stream consisting of one element.
    */
   def single[T](element: T): Source[T, Unit] = apply(SynchronousIterablePublisher(List(element), "single")) // FIXME optimize
+  
+  /**
+   * Create a `Source` that will continually emit the given element.
+   */
+  def constant[T](element: T): Source[T, Unit] = apply(() => Iterator.continually(element)) // FIXME optimize
```

Comment:
`continually` has at least in Scala collections the implication of repeated evaluation, while this is truly a constant value that is emitted. What about `repeat`?